compiler/luci/lang/include/luci/IR/Nodes/CircleRemoteConst.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_IR_CIRCLE_REMOTE_CONST_H__
+#define __LUCI_IR_CIRCLE_REMOTE_CONST_H__
+
+#include "luci/IR/CircleNodeDecl.h"
+#include "luci/IR/CircleOpcode.h"
+
+#include "luci/IR/CircleNodeMixins.h"
+
+#include <loco/IR/DataTypeTraits.h>
+
+namespace luci
+{
+
+/**
+ * @brief Class for remote access to provided tensor data
+ *
+ * @note  CircleRemoteConst has access to provided data and provides reading access to user.
+ * Node not owns given data, this is only interface to use data without memory allocation.
+ */
+class CircleRemoteConst final
```

Comment:
Please share why to introduce `CircleRemoteConst` ?

Reply:
To reduce memory usage by constants of model we can use constants directly from model's buffer. But `CircleConst` internally uses own `std::vector` for data and makes data copying during `import`/`export`. So, `CircleRemoteConst` is analogue of `CircleConst`, which operates data pointer instead own storage. 

----------------

**FYI** This design has one negative side: `CircleRemoteConst` not guarantees that provided data is valid. Following usage scenario is prohibited:

1. Create model's buffer.
2. Make import and bind data pointer to `CircleRemoteConst` during import.
3. Free model's buffer (**this action makes data given to `CircleRemoteConst` invalid**)
4. Use imported model. `CircleRemoteConst` will provide memory access to corrupted data.

Reply:
Thanks for the explanation. As above comment `Remote` seems ambigious.

As you wrote
```
3. Free model's buffer (this action makes data given to CircleRemoteConst invalid)
```
this can be critical for usage scenario. import routine should have self awared feature not to work for this case.

Reply:
> As above comment Remote seems ambigious.

@seanshpark What do you think about `CircleReferenceConst` instead `CircleRemoteConst` or may be `CircleConstProvider`?

Reply:
Um...  It would be better to explain why `Reference` or `Provider` as I can't find a clue why you used these words.
Please consider that I cannot deeply think on this topic as I am concentrating on other tasks.